### PR TITLE
Add support for pallet-xcm precompile in frontier template

### PR DESCRIPTION
**Pull request by Agusrodri**
### What does it do?

Adds the pallet-xcm precompile from Moonkit (built in https://github.com/Moonsong-Labs/moonkit/pull/35) to the frontier-template runtime.

### CODE REVIEW

File: container-chains/runtime-templates/frontier/src/precompiles.rs [hunk 1]
```diff
@@ -62,12 +67,22 @@ impl Erc20Metadata for NativeErc20Metadata {
...
+
 parameter_types! {
     pub ForeignAssetPrefix: &'static [u8] = FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX;
 }
 
 type EthereumPrecompilesChecks = (AcceptDelegateCall, CallableByContract, CallableByPrecompile);
 
+// Pallet-xcm precompile types.
+// The pallet-balances address is identified by ERC20_BALANCES_PRECOMPILE const
+type SingleAddressMatch = SingleAddressMatcher<AccountId, ERC20_BALANCES_PRECOMPILE, Balances>;
```
Comment (tmpolaczyk) [inline container-chains/runtime-templates/frontier/src/precompiles.rs:81]: This should have a name that mentions the balances precompile, like `BalancesPrecompileMatch` or similar

**Comment by Agusrodri** [comment]
I tested the migration against the `chopsticks_frontier_template_upgrade` zombie test and everything works fine.

### Before the upgrade (using runtime 600):

No code stored for pallet-xcm precompile address (`0x0000000000000000000000000000000000000804`).

Captura de pantalla 2024-06-04 a la(s) 12 33 59 a  m

### After the upgrade (using runtime 700):

Dummy code added successfully for address `0x0000000000000000000000000000000000000804`.

Captura de pantalla 2024-06-04 a la(s) 12 33 08 a  m

File: container-chains/runtime-templates/frontier/src/migrations.rs [hunk 2]
```diff
@@ -84,15 +132,18 @@ where
...
+            MigratePrecompileXcmDummyCode::<Runtime>(Default::default());
         vec![
             // Applied in runtime 400
             // Box::new(migrate_precompiles),
-            Box::new(migrate_polkadot_xcm_v1),
+/*             Box::new(migrate_polkadot_xcm_v1),
             Box::new(migrate_xcmp_queue_v2),
             Box::new(migrate_xcmp_queue_v3),
-            Box::new(migrate_xcmp_queue_v4),
+            Box::new(migrate_xcmp_queue_v4), */
```
Comment (Agusrodri) [inline container-chains/runtime-templates/frontier/src/migrations.rs:143]: Should we keep these migrations? I think the following ones were already applied in runtime 600:

- migrate_polkadot_xcm_v1
- migrate_xcmp_queue_v2
- migrate_xcmp_queue_v3
- migrate_xcmp_queue_v4

(correct me if I'm wrong)

Comment (girazoki) [reply container-chains/runtime-templates/frontier/src/migrations.rs:143]: yes we can remove them

**Review by girazoki — APPROVED** [review]
Approved but dont merge before 1.11.0

**Review by ParthDesai — APPROVED** [review]

**Review by fgamundi — APPROVED** [review]

**Pull request merged**